libcudacxx/test/libcudacxx/cuda/execution/execution_policy/get_stream.pass.cpp

```diff
@@ -69,6 +61,9 @@ void test()
   static_assert(!execution::__queryable_with<execution::unsequenced_policy, ::cuda::get_stream_t>);
 
   test(cuda::execution::__cub_par_unseq);
+
+  // Ensure that all works even if we have a memory resource
+  test(cuda::execution::__cub_par_unseq.set_memory_resource(::cuda::device_default_memory_pool(::cuda::device_ref{0})));
```

Comment:
Suggestion: I would intuitively name this:
```suggestion
  test(cuda::execution::__cub_par_unseq.with_memory_resource(::cuda::device_default_memory_pool(::cuda::device_ref{0})));
```

Reply:
I would use `with_xxx` for any kind of attachment. thrust uses `on(stream)`, which is also possible, but then `operator()` for the MR, which I don't think fits a new design.